misc(react-native): Remove outdated manual linking information (#10599)
getsentry/sentry-docs: docs/platforms/react-native/manual-setup/manual-setup.mdx

For more information about what options to use in `SENTRY_CLI_DEBUG_FILES_UPLOAD_EXTRA_ARGS` visit the [Sentry CLI Apple Debug Symbols](/platforms/apple/dsym/#sentry-cli) documentation.

> ### Using node with nvm or Volta

Reviewer: ```suggestion
### Using Node With nvm or Volta
```

> If you are using nvm or Volta, Xcode has trouble locating the default node binary, which can look similar to the [troubleshooting example](/platforms/react-native/troubleshooting/#using-node-with-nvm-or-volta).

Reviewer: ```suggestion
If you're using nvm or Volta, Xcode may have trouble locating the default node binary. The suggested workaround in this [troubleshooting example](/platforms/react-native/troubleshooting/#using-node-with-nvm-or-volta) can help.
```
Since we're removing the "how to" portion of this section, it helps to be more direct in sending them to troubleshooting docs :)
